drivers/src/lms.rs

```diff
@@ -453,11 +454,14 @@ impl Lms {
             temp = HashValue::<N>::from(digest);
             node_num /= 2;
             i += 1;
+            digest.0.fill(0);
         }
-        let candidate_key = temp;
+        let mut candidate_key = temp;
```

Comment:
I think you probably want to clear temp here too, no?

I suggest:

```
let candidate_key = &temp;
if *candidate_key != HashValue::from(lms_public_key.digest) {
    return Ok(false);
}
digest.0.fill(0);
temp.fill(0);
```

Reply:
Updated.

Reply:
Updated.